kctf-operator/pkg/controller/challenge/update/update.go

```diff
@@ -0,0 +1,270 @@
+// File that ensures if all configurations are correctly set
+// TODO: create errors in case we can't get the instance (error different from not found)
+package update
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	kctfv1alpha1 "github.com/google/kctf/pkg/apis/kctf/v1alpha1"
+	"github.com/google/kctf/pkg/controller/challenge/autoscaling"
+	"github.com/google/kctf/pkg/controller/challenge/deployment"
+	"github.com/google/kctf/pkg/controller/challenge/service"
+	"github.com/google/kctf/pkg/controller/challenge/volumes"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	netv1beta1 "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Check if the arrays of ports are the same
+func EqualPorts(found []corev1.ServicePort, wanted []corev1.ServicePort) bool {
+	if len(found) != len(wanted) {
+		return false
+	}
+
+	for i, _ := range found {
+		if found[i].Name != wanted[i].Name || found[i].Protocol != wanted[i].Protocol ||
+			found[i].Port != wanted[i].Port || found[i].TargetPort != wanted[i].TargetPort {
+			return false
+		}
+	}
+	return true
+}
+
+// Copy ports from one service to another
+func CopyPorts(found *corev1.Service, wanted *corev1.Service) {
+	found.Spec.Ports = []corev1.ServicePort{}
+	found.Spec.Ports = append(found.Spec.Ports, wanted.Spec.Ports...)
+}
+
+func UpdateNumReplicas(challenge *kctfv1alpha1.Challenge, currentReplicas *int32) bool {
```

Comment:
The thing is that we only use this updateNumReplicas inside the package, so I think it might not be very meaningful to just put numReplicas or is it? (I will correct the initial of some functions that only have internal use)

Reply:
yeah, I think you can leave it as it is